eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxo.hs

```diff
@@ -1,58 +1,244 @@
+import Cardano.Ledger.Binary (DecCBOR (..), EncCBOR (..), Sized (..))
+import Cardano.Ledger.Binary.Coders (
+  Decode (..),
+  Encode (..),
+  Wrapped (Open),
+  decode,
+  encode,
+  (!>),
+  (<!),
+ )
+import Cardano.Ledger.Coin (Coin)
+import Cardano.Ledger.Conway.Core
+import Cardano.Ledger.Conway.Era (ConwayEra, ConwayUTXO, ConwayUTXOS)
+import Cardano.Ledger.Conway.Rules.Utxos (
+  ConwayUtxosPredFailure (..),
+ )
+import Cardano.Ledger.Conway.Tx (AlonzoTx (..))
+import Cardano.Ledger.Plutus (ExUnits)
+import Cardano.Ledger.Rules.ValidationMode (runTest, runTestOnSignal)
+import Cardano.Ledger.Shelley.API.Types (Network)
+import Cardano.Ledger.Shelley.LedgerState (UTxOState (utxosUtxo))
+import Cardano.Ledger.Shelley.Rules (
+  ShelleyUtxoPredFailure,
+  UtxoEnv,
+ )
+import qualified Cardano.Ledger.Shelley.Rules as Shelley
+import Cardano.Ledger.TxIn (TxIn)
+import Cardano.Ledger.UTxO (EraUTxO, UTxO (..))
+import Control.DeepSeq (NFData)
+import Control.Monad.Trans.Reader (asks)
+import Control.State.Transition.Extended (
+  Embed (..),
+  STS (..),
+  TRC (..),
+  TransitionRule,
+  judgmentContext,
+  liftSTS,
+  trans,
+  validate,
+ )
+import Data.Coerce (coerce)
+import Data.List.NonEmpty (NonEmpty)
+import Data.Set (Set)
+import GHC.Generics (Generic)
+import GHC.Natural (Natural)
+import Lens.Micro
+
+data ConwayUtxoPredFailure era
+  = -- | The bad transaction inputs
+    BadInputsUTxO
+      !(Set (TxIn (EraCrypto era)))
+  | OutsideValidityIntervalUTxO
+      -- | transaction's validity interval
+      !ValidityInterval
+      -- | current slot
+      !SlotNo
+  | MaxTxSizeUTxO
+      -- | the actual transaction size
+      !Integer
+      -- | the max transaction size
+      !Integer
+  | InputSetEmptyUTxO
+  | FeeTooSmallUTxO
+      -- | the minimum fee for this transaction
+      !Coin
+      -- | the fee supplied in this transaction
+      !Coin
+  | ValueNotConservedUTxO
+      -- | the Coin consumed by this transaction
+      !(Value era)
+      -- | the Coin produced by this transaction
+      !(Value era)
+  | -- | the set of addresses with incorrect network IDs
+    WrongNetwork
+      -- | the expected network id
+      !Network
+      -- | the set of addresses with incorrect network IDs
+      !(Set (Addr (EraCrypto era)))
+  | WrongNetworkWithdrawal
+      -- | the expected network id
+      !Network
+      -- | the set of reward addresses with incorrect network IDs
+      !(Set (RewardAccount (EraCrypto era)))
+  | -- | list of supplied transaction outputs that are too small
+    OutputTooSmallUTxO
+      ![TxOut era]
+  | -- | Subtransition Failures
+    UtxosFailure (PredicateFailure (EraRule "UTXOS" era))
+  | -- | list of supplied bad transaction outputs
+    OutputBootAddrAttrsTooBig
+      ![TxOut era]
+  | -- | list of supplied bad transaction output triples (actualSize,PParameterMaxValue,TxOut)
+    OutputTooBigUTxO
+      ![(Integer, Integer, TxOut era)]
```

Comment:
Let's change those to an `Int` instead. There is really no reason for an arbitrary precision `Integer` in here. Note that I added a point to #4085 about doing the same for previous eras, but we'll do that past Conway. This means that you'll have some conversion from `Integer` -> `Int` during injection. Make sure to make a note about this upcoming change and make the conversion monophonic with `fromIntegral :: Integer -> Int`. This way once we change the `OutputTooBigUTxO` from the previous era we will not forget to remove this note and the conversion.
```suggestion
      ![(Int, Int, TxOut era)]
```